test/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyControllerSpec.scala

```diff
@@ -280,6 +282,272 @@ class SubsidyControllerSpec
+            )
+          }
+
+          val result = performAction(
+            "report-payment-return" -> "false"
+          )
+
+          status(result) shouldBe SEE_OTHER
+          redirectLocation(result) should contain(routes.NoClaimNotificationController.getNoClaimNotification.url)
+        }
+      }
+    }
+
+    "handling request to get returning user page" must {
+
+      def performAction = controller.getReportedPaymentReturningUserPage(
+        FakeRequest(GET, routes.SubsidyController.getReportedPaymentReturningUserPage.url)
+      )
+
+      "call to get session fails" in {
+        inSequence {
+          mockAuthWithEnrolmentAndValidEmail()
+          mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+          mockGetOrCreate[SubsidyJourney](eori1)(Left(ConnectorError(exception)))
+        }
+        assertThrows[Exception](await(performAction))
+      }
+
+      "display the page" when {
+        "a request is made" in {
+          inSequence {
+            mockAuthWithEnrolmentAndValidEmail()
+            mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+            mockGetOrCreate[SubsidyJourney](eori1)(Right(subsidyJourney))
+          }
+          checkPageIsDisplayed(
+            performAction,
+            messageFromMessageKey("reportPaymentReturnJourney.title"),
+            { doc =>
+              doc.select(".govuk-back-link").attr("href") shouldBe routes.AccountController.getAccountPage.url
+              doc
+                .select("form")
+                .attr("action") shouldBe routes.SubsidyController.postReportedPaymentReturningUserPage.url
+            }
+          )
+        }
+      }
+    }
+
+    "handling get to reported no custom subsidy page" must {
+      def performAction(): Future[Result] = controller.getReportedNoCustomSubsidyPage(FakeRequest())
+
+      "render the page with updated form and correct URLs" in {
+        val journeyWithReportedNonCustomSubsidyReturningUser = subsidyJourney
+          .setReportedPaymentReturningUser(true)
+
+        inSequence {
+          mockAuthWithEnrolmentAndValidEmail(eori1)
+          mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+          mockGetOrCreate[SubsidyJourney](eori1)(Right(journeyWithReportedNonCustomSubsidyReturningUser))
+        }
+
+        val result = performAction()
+
+        status(result) shouldBe OK
+        contentAsString(result) should include(
+          messageFromMessageKey("reportNonCustomSubsidy.title")
+        )
+        contentAsString(result) should include(routes.SubsidyController.getReportedPaymentReturningUserPage.url)
+        contentAsString(result) should not include routes.SubsidyController.getReportPaymentFirstTimeUser.url
+        contentAsString(result) should include(LocalDate.now.getYear.toString)
+        contentAsString(result) should include((LocalDate.now.getYear - 1).toString)
```

Comment:
there is some logic in those <li>'s we should test the generated string e.g. "6 April 2020 to 5 April 2021" rather than just searching the page for numbers. We should select each <li> and check the text